department-of-veterans-affairs/vets-website
# Adds ctaWidget for supply reordering

## Summary

- Adds the ability to feature toggle the ctaWidget. 
- Adds ctaWidget app-id: `ha-cpap-supplies`
- Feature toggle for app-id: `ha_cpap_supplies_cta`
- Team: #mhv-on-vagov-cartography-team
 
## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#79823
- _Link to epic if not included in ticket_
department-of-veterans-affairs/va.gov-team#77499

## Testing done

Steps to verify in local development:

1. `yarn build` from `vets-website`
1. `yarn watch` from `content-build`
1. `foreman start -m all=1,clamd=0,freshclam=0` from `vets-api` branch `radavis/79823-ha-cpap-cta`
1. Add HTML to `build/localhost/health-care/order-hearing-aid-batteries-and-accessories/index.html`: `<div data-template="paragraphs/react_widget" class="cta-widget" data-widget-type="cta" data-app-id="ha-cpap-supplies"></div>` Note that this step is done through the Drupal content editor for other environments.
1. Visit: `http://localhost:3002/health-care/order-hearing-aid-batteries-and-accessories/#how-do-i-order-hearing-aid-bat`, observe that no content has changed.
1. Visit: `http://localhost:3000/flipper/features/ha_cpap_supplies_cta` and click "Enable for everyone"
1. Visit: `http://localhost:3002/health-care/order-hearing-aid-batteries-and-accessories/#how-do-i-order-hearing-aid-bat`, observe that content _has_ changed.

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Desktop |   ![Screenshot 2024-04-30 at 13-45-21 Order hearing aid batteries and accessories from VA Veterans Affairs](https://github.com/department-of-veterans-affairs/vets-website/assets/1299034/d3342208-bec7-4e09-9849-7b0407ed7baa)     |   ![Screenshot 2024-04-30 at 13-45-54 Order hearing aid batteries and accessories from VA Veterans Affairs](https://github.com/department-of-veterans-affairs/vets-website/assets/1299034/ed9947aa-5d84-410d-8c98-2e3629bd32e9)    |

## Acceptance criteria

### Quality Assurance & Testing
